File: design-patterns/JavaDesignPatterns/src/com/bigboxcode/designpattern/factory/transportmethod/RoadTransportFactory.java
package com.bigboxcode.designpattern.factory.transportmethod;

public class RoadTransportFactory extends TransportFactory {
    @Override
    public Transport getTransport(String name) {
        if (name.equalsIgnoreCase("car")) {
            return new Car();
        }

        if (name.equalsIgnoreCase("bike")) {
            return new Bike();
        }

        if (name.equalsIgnoreCase("bus")) {
            return new Bus();
        }

        return null;
    }
}

File: design-patterns/php/factory/transportmethod/RoadTransportFactory.php
<?php
// RoadTransportFactory.php

namespace BigBoxCode\DesignPattern\Factory\TransportMethod;

class RoadTransportFactory extends TransportFactory {
    public function getTransport(string $name): ?Transport {
        switch (strtolower($name)) {
            case "car":
                return new Car();
            case "bike":
                return new Bike();
            case "bus":
                return new Bus();
        }

        return null;
    }
}